docroot/profiles/custom/sitenow/sitenow.install

```diff
@@ -3533,3 +3533,474 @@ function sitenow_update_10016() {
+    },
+    'component_group_size' => function ($section) {
+      return (count($section->getComponents()) === 4) ? 4 : 3;
+    },
+    'section_generator' => function ($args) {
+      $count = count($args['components']);
+      $layout_id = ($count === 4) ? 'layout_fourcol_section' : 'layout_threecol_section';
+      $section = new Section($layout_id);
+      $layout_settings = $args['current_section']->getLayoutSettings();
+      if ($layout_id == 'layout_threecol_section') {
+        $layout_settings['column_widths'] = '33-34-33';
+      }
+      $styles = $layout_settings['layout_builder_styles_style'] ?? [];
+
+      // Move over all settings except for the grid style.
+      // Remove only the grid styles, keep the others.
+      $styles = array_filter($styles, function ($style) {
+        return !in_array($style, [
+          'section_layout_grid_3x2_stacked',
+          'section_layout_grid_3x2',
+        ], TRUE);
+      });
+
+      // Make sure the following styles are included in the new section.
+      $required_styles = [
+        'section_margin_remove_default_margins',
+        'section_padding_remove_default_padding',
+      ];
+      foreach ($required_styles as $style) {
+        if (!in_array($style, $styles, TRUE)) {
+          $styles[] = $style;
+        }
+      }
+      $layout_settings['layout_builder_styles_style'] = $styles;
+
+      $section->setLayoutSettings($layout_settings);
+
+      // Add one component to each region.
+      $regions_by_layout = [
+        'layout_fourcol_section' => ['first', 'second', 'third', 'fourth'],
+        'layout_threecol_section' => ['first', 'second', 'third'],
+      ];
+      $regions = $regions_by_layout[$layout_id];
+      foreach ($args['components'] as $i => $component) {
+        $section->appendComponent($component->setRegion($regions[$i] ?? 'first'));
+      }
+      return $section;
+    },
+    'post_process' => function ($data) {
+      $top_styles = ['remove_default_top_padding', 'section_padding_top_extra'];
+      $bottom_styles = ['remove_default_bottom_padding', 'section_padding_bottom_extra'];
+      $last_index = count($data['sections']) - 1;
+
+      foreach ($data['sections'] as $index => $section) {
+        $settings = $section->getLayoutSettings();
+        $styles = (array) ($settings['layout_builder_styles_style'] ?? []);
+
+        // Remove top styles from everything except the first.
+        if ($index !== 0) {
+          $styles = array_diff($styles, $top_styles);
+        }
+
+        // Remove bottom styles from everything except the last.
+        if ($index !== $last_index) {
+          $styles = array_diff($styles, $bottom_styles);
+        }
+
+        $settings['layout_builder_styles_style'] = array_values($styles);
+        $section->setLayoutSettings($settings);
+      }
+    },
+  ]);
+}
+
+/**
+ * Remove deprecated layout builder styles from all sections.
+ */
+function sitenow_update_10019() {
```

Comment:
Is this one necessary if the config styles are removed?

Reply:
Wondering because in https://github.com/uiowa/uiowa/pull/8484 we removed deprecated styles from the banner config but did not remove them in an update hook like this.

Reply:
i noticed there was a style still in tempstore for the sandbox cta four column example and just assumed the same for the other tables. i'd be fine with getting rid of it if we aren't worried about it

Reply:
Removed it https://github.com/uiowa/uiowa/pull/8669/commits/67cc2aa79ed873610be2595dbcfc3284e021c33d